### PR TITLE
Add compact comparer

### PR DESCRIPTION
**Pull request by asherber**
I don't know if this is something you might be interested in or not. As noted in VerifyTests/Verify#425, I often have verified files that are a couple hundred lines long. Both the default output from Verify and the default output from Verify.DiffPlex wind up being very long and difficult to sift through.

What I've done is add a "compact" comparer that only outputs the actual changed lines, plus one line of context before and after. You can see the output [here](https://github.com/asherber/Verify.DiffPlex/blob/compact-output/src/Tests/Tests.Compact.verified.txt). I find this splits the difference between the default Verify.DiffPlex and the other solution I came up with using `Assert.Equals()`, which fails fast and only shows the first difference.

The compact comparer is used by calling `VerifyDiffPlex.Initialize(compact: true)`, though that could be changed to something like `VerifyDiffPlex.InitializeCompact()`. I'm happy to revise as needed.

### CODE REVIEW

**Comment by SimonCropp** [comment]
sorry for the delayed reply.

can you add some documentation for this feature?

**Comment by asherber** [comment]
Sure -- do you want it just in the REAMDE for this repo?

**Comment by SimonCropp** [comment]
yep. in the readme is fine

**Comment by asherber** [comment]
Changing this to a draft -- I have a few edge cases I want to work through.

**Comment by asherber** [comment]
Dealt with edge cases, taking PR out of draft status.

I've made 3 changes:

1.  If first or last line of file is changed, display `[BOF]` or `[EOF]` as context line
2.  Handle the case where two change sections are separated by a single unchanged line
3.  Reduce leading whitespace

Example of no.2:

```
5 Unchanged line
- Before
+ After
7 Another unchanged line
- Another before
+ Another after
9 Still another unchanged line
```

For no.3, the +/- symbols are now right-aligned with the line numbers for unchanged lines, instead of being further indented. You can see above what happens with files with fewer than 10 lines. If the file has more than 10 but fewer than 100 lines:

```
// Original version
53   Unchanged line
   - Before
   + After
55   Unchanged line

// New version
53 Unchanged line
 - Before
 + After
55 Unchanged line
```

Seems to me a little tidier and easier to read.

**Comment by asherber** [comment]
@SimonCropp Can you please review?

File: src/Tests/Tests.Compact.verified.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ + Line 7 changed
+09 Line 8
+
+11 Line 10
+ - Line 11
+ - Line 12
+ + Line 11 changed
+ + Line 12 changed
+   [EOF]
+
```
Comment (SimonCropp) [inline src/Tests/Tests.Compact.verified.txt:29]: why the trailing whitespace?

Comment (asherber) [reply src/Tests/Tests.Compact.verified.txt:29]: This isn't anything I'm doing -- I just stepped through the test, and nothing gets added in my code after the \[EOF\]. If you look at [Tests.Simple.Verified.txt](https://github.com/VerifyTests/Verify.DiffPlex/blob/main/src/Tests/Tests.Simple.verified.txt), it also has three blank lines at the end, so I'm guessing it's something in the Verify plumbing.

**Comment by SimonCropp** [comment]
this is now deployed

**Pull request merged**